test/jdk/java/util/zip/ZipInputStream/Zip64DataDescriptor.java

```diff
@@ -87,8 +95,55 @@ public void shouldReadZip64Descriptor() throws IOException {
                 0000b20000000000000001000000504b050600000000010001005c000000
                 560000000000""";
 
-        byte[] zip = HexFormat.of().parseHex(hex.replaceAll("\n", ""));
+        zip = HexFormat.of().parseHex(hex.replaceAll("\n", ""));
+    }
+
+    /**
+     * Verify that small-sized Zip64 entries can be parsed by ZipInputStream
+     */
+    @Test
+    public void shouldReadZip64Descriptor() throws IOException {
+        readZipInputStream();
+    }
+
+    /**
+     * For maximal backward compatibility when reading Zip64 descriptors, invalid
+     * Zip64 extra data sizes should be ignored
+     */
+    @Test
+    public void shouldIgnoreInvalidExtraSize() {
+
+        setExtraSize((short) 42);
+
+        ZipException ex = expectThrows(ZipException.class, () -> {
+            readZipInputStream();
+        });
+
+        assertEquals(ex.getMessage(), "invalid entry size (expected 0 but got 5 bytes)");
+    }
+
+    /**
+     * Validate that an extra size exceeding the extra size is ignored
+     */
+    @Test
+    public void shouldIgnoreExcessiveExtraSize() {
+
+        setExtraSize(Short.MAX_VALUE);
+
+        ZipException ex = expectThrows(ZipException.class, () -> {
+            readZipInputStream();
+        });
+
+        assertEquals(ex.getMessage(), "invalid entry size (expected 0 but got 5 bytes)");
+    }
+
+    private void setExtraSize(short invalidSize) {
```

Comment:
Please add a comment describing the method (and any others without).  As we add new tests, we are trying to. follow this practice the best that we can.

Reply:
Added comments for these methods and also for the zip field which I renamed to zip64Zip.